addons/point_of_sale/models/pos_session.py

```diff
@@ -185,31 +189,50 @@ def create(self, values):
         # the .xml files as the CoA is not yet installed.
         pos_config = self.env['pos.config'].browse(config_id)
         ctx = dict(self.env.context, company_id=pos_config.company_id.id)
+        if not pos_config.journal_id:
```

Comment:
you can remove this block in create. 'pos_config.journal_id' and 'pos_config.payment_method_ids' are now required and should be correctly set on config